### PR TITLE
feat(workflow v2): uploadArtifact

### PR DESCRIPTION
**Pull request by fsamin**

### CODE REVIEW

File: contrib/grpcplugins/action/plugin-uploadArtifact/plugin-uploadArtifact.yml [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+author: "François SAMIN <francois.samin@corp.ovh.com>"
+description: |
+  This uploads artifacts from your workflow allowing you to share data between jobs and store data once a workflow is complete.
+inputs:
+  path:
+    type: text
+    description: A file, directory or wildcard pattern that describes what to upload
+    required: true
+  if-no-files-found:
+    description: >
```
Comment (sguiheux) [inline contrib/grpcplugins/action/plugin-uploadArtifact/plugin-uploadArtifact.yml:12]: missing type ?

File: engine/api/v2_workflow_run_engine.go [hunk 2]
```diff
@@ -273,6 +281,22 @@ func prepareRunJobs(ctx context.Context, run sdk.V2WorkflowRun, wrEnqueue sdk.V2
 				if jobDef.WorkerModel != "" {
 					runJob.ModelType = run.WorkflowData.WorkerModels[jobDef.WorkerModel].Type
 				}
+
+				if !jobDef.Integrations.IsEmpty() {
```
Comment (sguiheux) [inline engine/api/v2_workflow_run_engine.go:285]: wrong place

**Review by sguiheux — CHANGES_REQUESTED** [review]

**Comment by ovh-cds** [comment]
CDS Report build-all-cds#19346.0 ✘  
* Build
  * Build and Package All ✔ 
  * Build Debpacker ✔ 
  * Build SMTPMock ✔ 
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ✔ 
  * Test SDK ✔ 
  * Test UI ✔ 
* Package
  * Docker ✔ 
* Integration
  * Docker Compose Tests ✔ 
  * Standalone Tests ✘

File: engine/api/v2_workflow_run_engine.go (hunk 2 above)
Comment (fsamin) [reply engine/api/v2_workflow_run_engine.go]: done

**Comment by ovh-cds** [comment]
CDS Report build-all-cds#19348.0 ✘  
* Build
  * Build and Package All ✔ 
  * Build Debpacker ✔ 
  * Build SMTPMock ✔ 
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ✔ 
  * Test SDK ✔ 
  * Test UI ✔ 
* Package
  * Docker ✔ 
* Integration
  * Docker Compose Tests ✔ 
  * Standalone Tests ✘

**Comment by ovh-cds** [comment]
CDS Report build-all-cds#19350.0 ✘  
* Build
  * Build and Package All ✔ 
  * Build Debpacker ✔ 
  * Build SMTPMock ✔ 
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ✔ 
  * Test SDK ✔ 
  * Test UI ✔ 
* Package
  * Docker ✔ 
* Integration
  * Docker Compose Tests ✔ 
  * Standalone Tests ✘

**Comment by ovh-cds** [comment]
CDS Report build-all-cds#19350.1 ✘  
* Build
  * Build and Package All ✔ 
  * Build Debpacker ✔ 
  * Build SMTPMock ✔ 
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ✔ 
  * Test SDK ✔ 
  * Test UI ✔ 
* Package
  * Docker ✔ 
* Integration
  * Docker Compose Tests ✔ 
  * Standalone Tests ✘

**Review by sguiheux — APPROVED** [review]

**Comment by ovh-cds** [comment]
CDS Report build-all-cds#19353.0 ✘  
* Build
  * Build and Package All ✔ 
  * Build Debpacker ✔ 
  * Build SMTPMock ✘  
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ✔ 
  * Test SDK ✘  
  * Test UI ✔

**Comment by ovh-cds** [comment]
CDS Report build-all-cds#19355.0 ✘  
* Build
  * Build and Package All ✔ 
  * Build Debpacker ✔ 
  * Build SMTPMock ✔ 
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ✘  
  * Test SDK ✔ 
  * Test UI ✔

**Review by yesnault — APPROVED** [review]

**Comment by ovh-cds** [comment]
CDS Report build-all-cds#19358.0 ✘  
* Build
  * Build and Package All ✔ 
  * Build Debpacker ✔ 
  * Build SMTPMock ✘  
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ✘  
  * Test SDK ✔ 
  * Test UI ✔

**Pull request merged**